controller/Controller.class.php

```diff
@@ -148,13 +148,24 @@ protected static function getRouterWithRoutes(): \Routing\RouteCollector
                 Response::enableHttpCache();
                 return ['page/' . $slug, []];
             } else {
-                return NavActions::execute404();
+                return static::doLowerUriRedirectOr404($slug);
             }
         });
 
         return $router;
     }
 
+    protected static function doLowerUriRedirectOr404($uri, $router = null)
+    {
+        $lowerUri = strtolower($uri);
+        if ($uri !== $lowerUri) {
+            if (View::exists('page/' . $lowerUri) || isset($router) && $router.hasroute($lowerUri)) {
```

Comment:
Did you run this? I don't think `$router.hasroute` is valid PHP

Reply:
yes,
In route_collector.class.php which is the base class for $router:
```
public function hasRoute(string $name): bool
    {
        return isset($this->reverse[$name]);
    }
```

Reply:
So you're confidant this code works?

Reply:
yes, I tested 404, dynamic and static routes.